plugin/evm/gossiper.go

```diff
@@ -270,34 +282,49 @@ func (n *pushGossiper) awaitEthTxGossip() {
 				}
 			case <-priorityRegossipTicker.C:
 				for _, tx := range n.queuePriorityRegossipTxs() {
-					n.txsToGossip[tx.Hash()] = tx
+					n.ethTxsToGossip[tx.Hash()] = tx
 				}
-				if attempted, err := n.gossipTxs(true); err != nil {
+				if attempted, err := n.gossipEthTxs(true); err != nil {
 					log.Warn(
 						"failed to regossip priority eth transactions",
 						"len(txs)", attempted,
 						"err", err,
 					)
 				}
-			case txs := <-n.txsToGossipChan:
+			case txs := <-n.ethTxsToGossipChan:
 				for _, tx := range txs {
-					n.txsToGossip[tx.Hash()] = tx
+					n.ethTxsToGossip[tx.Hash()] = tx
 				}
-				if attempted, err := n.gossipTxs(false); err != nil {
+				if attempted, err := n.gossipEthTxs(false); err != nil {
 					log.Warn(
 						"failed to send eth transactions",
 						"len(txs)", attempted,
 						"err", err,
 					)
 				}
+
+				gossipTxs := make([]*GossipEthTx, 0, len(txs))
+				for _, tx := range txs {
+					gossipTxs = append(gossipTxs, &GossipEthTx{Tx: tx})
+				}
+
+				n.ethTxGossiper.Add(gossipTxs...)
+				if err := n.ethTxGossiper.Gossip(context.TODO()); err != nil {
```

Comment:
hmm good point, I ported changes from Coreth directly. cc @joshua-kim

Reply:
The only reason this was left as TODO was to minimize the diff... before we were using the `p2p.Network` code the old coreth code used `TODO` as well: https://github.com/ava-labs/coreth/blob/347fe530aab2c95103ee48e23b9ffaff986a0ca5/peer/network.go#L193. We should be using timed contexts as per @marun 's comment, but it was simply not added in because it made the diff bigger while being equivalent to the previous code.

Reply:
created https://github.com/ava-labs/subnet-evm/issues/1050